src/glue.c

```diff
@@ -475,16 +475,6 @@ void genObjFile(Module *m, bool multiobj)
         m->ssharedctor = callFuncsAndGates(m, &ssharedctors, (StaticDtorDeclarations *)&esharedctorgates, "__modsharedctor");
         m->sshareddtor = callFuncsAndGates(m, &sshareddtors, NULL, "__modshareddtor");
         m->stest = callFuncsAndGates(m, &stests, NULL, "__modtest");
-
-        if (m->doppelganger)
-            genModuleInfo(m);
-    }
-
-    if (m->doppelganger)
-    {
-        objc_Module_genmoduleinfo_classes();
```

Comment:
No. The change is to always call `genModuleInfo`. `objc_Module_genmoduleinfo_classes` is called by `genModuleInfo`, I removed some of the duplications as well. Seems like the bigger if statement above didn't catch all cases where the module info needed to be generated. `m->doppelganger` is `true` if the `-lib` flag is passed. Before my change the function would return, if the `-lib` flag was passed, before the module info had been generated.